src/Extension/FluentSiteTreeExtension.php

```diff
@@ -366,6 +402,158 @@ protected function updateSavePublishActions(FieldList $actions)
         }
     }
 
+    /**
+     * Update publish action state to reflect the localised record instead of the base record
+     *
+     * @param FieldList $actions
+     */
+    protected function updatePublishState(FieldList $actions): void
+    {
+        $owner = $this->owner;
+
+        if (!$owner->isInDB()) {
+            return;
+        }
+
+        $published = $owner->isPublishedInLocale();
+
+        if (!$published) {
+            return;
+        }
+
+        /** @var CompositeField $majorActions */
+        $majorActions = $actions->fieldByName('MajorActions');
+
+        if (!$majorActions) {
+            return;
+        }
+
+        $publishAction = $majorActions->fieldByName('action_publish');
+
+        if (!$publishAction) {
+            return;
+        }
+
+        // make sure that changes only on the base record
+        // do not trigger "need to publish" button state
+        // this is needed because the default interface looks
+        // at the base record instead of the localised page
+        $publishAction
+            ->setTitle(_t(SiteTree::class . '.BUTTONPUBLISHED', 'Published'))
```

Comment:
```suggestion
            ->setTitle(_t('SilverStripe\\CMS\\Model\\SiteTree.BUTTONPUBLISHED', 'Published'))
```

`SiteTree::class` won't work in the text collector unfortunately